components/core/src/os/net/mod.rs

```diff
@@ -22,3 +22,73 @@ mod imp;
 mod imp;
 
 pub use self::imp::*;
+
+extern crate dns_lookup;
+
+use dns_lookup::{getaddrinfo,
+                 AddrInfoHints};
+use std::io;
+
+// lookup_fqdn returns a vector of fqdn that resolves the provided hostname.
+//
+// Since the underlying crate is platform agnostic, this function is as well,
+// we only implement a single variable `flags` that has the right hint for
+// the running operating system.
+pub fn lookup_fqdn(hostname: &str) -> io::Result<String> {
+    #[cfg(not(windows))]
+    let flags = libc::AI_CANONNAME;
+
+    #[cfg(windows)]
+    let flags = winapi::shared::ws2def::AI_CANONNAME;
+
+    let hints = AddrInfoHints { flags,
+                                ..AddrInfoHints::default() };
+    let addrinfos =
+        getaddrinfo(Some(hostname), None, Some(hints))?.collect::<std::io::Result<Vec<_>>>()?;
+
+    // If 'hints.ai_flags' includes the AI_CANONNAME flag, then the ai_canonname
+    // field of the first of the addrinfo structures in the returned list is set
+    // to point to the official name of the host.
+    if !addrinfos.is_empty() {
+        let addrinfo = addrinfos[0].clone();
+        return Ok(addrinfo.canonname.unwrap_or_else(|| hostname.to_string()));
+    }
+
+    Ok(hostname.to_string())
+}
+
+// fqdn returns the fully qualified domain name of the running machine
+pub fn fqdn() -> Option<String> {
+    match hostname() {
```

Comment:
Interesting... the `dns_lookup` crate has its own `get_hostname` function... might be a good opportunity to get rid of some more code 😄

Reply:
Done done done!!!! Really simple and great implementation. 🥇